### DISCUSSION TITLE
Place custom polyfills in polyfills.js instead of main.js?

### DISCUSSION THREAD

**Discussion by Tylerian** [vercel/next.js#14069] [Help] [answered]
According to the official examples, it is best practice to add your custom polyfills to Webpack's `main.js` entry point instead of `polyfills.js`... Is there any specific reason for doing so?

This is [the example](https://github.com/vercel/next.js/blob/86160a5190c50ea315c7ba91d77dfb51c42bc65f/examples/with-polyfills/next.config.js) i'm referring to.

As a side note, I find it very cumbersome (especially for newbies) to modify the Webpack config in order to add polyfills. Perhaps Next.js could offer a simpler way to do this? Something like an optional file `next-polyfills.js` at the root of the project just like it does with `next-env.d.ts`  for TypeScript typings.

**Comment by jamesmosier** [comment]
Next offers built-in polyfills as of 9.3. The list [can be found here](https://github.com/vercel/next.js/blob/canary/packages/next-polyfill-nomodule/src/index.js)

**Reply by Tylerian** [reply]
Hey thanks for the answer, unfortunately that's not what I'm asking for.

**Comment by Tylerian** [comment, marked as answer]
Answering myself as I dug more into this...

Turns out the [with-polyfills](https://github.com/vercel/next.js/blob/86160a5190c50ea315c7ba91d77dfb51c42bc65f/examples/with-polyfills) example is known to cause issues with bundling.

The proper way (and best practice) to add custom polyfills is to include them as the first line into `pages/_app.js`. (source: #13766 )
